### PR TITLE
feature/storybook7

### PR DESCRIPTION
**Pull request by tpoisseau**
Hello,

To test my new setup, I updated this repo to storybook 7 with vite builder.
I kept cjs and esm builds but due to some restrictions from vite toolchain I had to change paths and name of build files.
I updated the binding in package.json but I don't know if it should be marked as breaking change.

Best regards

### CODE REVIEW

**Review by stropitek — COMMENTED** [review]
SGTM

I think you can remove the babel dependencies.

**Comment by tpoisseau** [comment]
Good point, I forgot to commit after cleanup. It's good now.

I let you squash my commits and merge if it's OK for you.

File: package.json [hunk 1]
```diff
@@ -8,25 +8,23 @@
...
   ],
-  "main": "lib/index.js",
+  "main": "lib/index.cjs.js",
   "keywords": [
     "react",
     "molecular formula",
     "mf renderer"
   ],
-  "module": "lib-es6/index.js",
+  "module": "lib/index.es.js",
```
Comment (targos) [inline package.json]: Is it possible to remove the `.js` suffix?

Comment (targos) [reply package.json]: I mean in the actual filename, not just here.

File: package.json [hunk 2]
```diff
@@ -36,20 +34,24 @@
...
+    "@storybook/addon-storysource": "^7.0.4",
+    "@storybook/react": "^7.0.4",
+    "@storybook/react-vite": "^7.0.4",
+    "@types/react": "^17.0.2",
+    "@types/react-dom": "^17.0.2",
+    "@vitejs/plugin-react": "^3.1.0",
     "cross-env": "^7.0.3",
     "eslint": "^8.29.0",
     "eslint-config-zakodium": "7.0.0",
+    "eslint-plugin-storybook": "^0.6.11",
```
Comment (targos) [inline package.json:48]: Is this used even if we don't extend it in the eslint config?

File: package.json (hunk 1 above)
Comment (targos) [reply package.json]: If possible, then we should have `index.cjs` and `index.mjs`.

Comment (tpoisseau) [reply package.json]: I did not found a solution to do it with one run. So I took the same approach you had with babel :
Two build with an env variable to tweak the build config.

File: package.json (hunk 2 above)
Comment (tpoisseau) [reply package.json:48]: It has been auto-added from the storybook migrator.
Now it's added in eslint config of project.

**Review by targos — CHANGES_REQUESTED** [review]
Good work!

A few issues:
- `npm run build-storybook` gives an error: `Invalid value for option "preserveEntrySignatures" - setting this option to false is not supported for "output.preserveModules".`
- I think we should disable minification in the `lib` build. We usually don't publish minified code to npm (harder to debug).

**Comment by tpoisseau** [comment]
And yet another env variable for build-storybook.
I'm not sure it's the best solution but it seems work.

And yes, vite minify by default, I disabled it.

**Review by targos — APPROVED** [review]

**Comment by targos** [comment]
Unfortunately, it seems to have broken something in the build: https://github.com/zakodium-oss/react-mf/issues/17

**Pull request merged**